Towny/src/main/java/com/palmergames/bukkit/towny/object/Transaction.java

```diff
@@ -1,24 +1,52 @@
 package com.palmergames.bukkit.towny.object;
 
+import org.bukkit.Bukkit;
 import org.bukkit.entity.Player;
+import org.jetbrains.annotations.Nullable;
+
+import com.palmergames.bukkit.towny.object.economy.Account;
 
 public class Transaction {
 	private final TransactionType type;
-	private final Player player;
+	private final Account receivingAccount;
+	private final Account sendingAccount;
 	private final double amount;
 	
-	public Transaction(TransactionType type, Player player, double amount) {
+	public Transaction(TransactionType type, Account sendingAccount, Account receivingAccount, double amount) {
 		this.type = type;
-		this.player = player;
+		this.sendingAccount = sendingAccount;
+		this.receivingAccount = receivingAccount;
 		this.amount = amount;
 	}
 
 	public TransactionType getType() {
 		return type;
 	}
 
+	public String getName() {
```

Comment:
The getName and getTownyObject methods are very ambigious and probably not needed

Reply:
Nuked them, they weren't being used in this iteration and they weren't already part of Transaction.